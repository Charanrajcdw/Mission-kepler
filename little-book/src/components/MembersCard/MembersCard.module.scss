.memberCard {
  margin-bottom: 36px;

  .memberName {
    color: var(--member-name);
    font-size: 0.97rem;
    padding-top: 6px;
  }

  .memberCity {
    color: var(--member-city);
    font-size: 0.88rem;
  }
}
