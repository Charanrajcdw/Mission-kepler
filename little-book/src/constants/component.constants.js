export const MODAL = {
  members: "Members",
  newBlog: "Add New Blog",
  remove: "Remove",
};

export const MEMBERS_LIST = {
  title: "Members",
  noMembers: "No members available",
};

export const BLOG_FORM = {
  title: "Add New Blog",
  button: "ADD",
  titlePlaceholder: "Name your blog",
  detailsPlaceholder: "Write Content Here ..",
  invalidValues: "Invalid title or description",
  invalidBlog: "Blog title already exists",
  blogAdded: "New Blog added successfully",
  localBlog: "local",
};

export const WARNING_MODAL = {
  title: "Unsaved changes will be lost. Are you sure to continue?",
  continue: "Continue",
  cancel: "Cancel",
};
