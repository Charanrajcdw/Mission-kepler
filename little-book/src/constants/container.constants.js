export const SIDEBAR = {
  logo: "Little Book",
  filter: "FILTER",
  noFilter: "Add blogs to show filters",
  viewMembers: "View Members",
  switchToDark: "Switch to Dark Mode",
  switchToLight: "Switch to Light Mode",
  blogs: "Blogs",
};
