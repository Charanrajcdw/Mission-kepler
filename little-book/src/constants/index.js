export * from "./url.constants";
export * from "./theme.constants";
