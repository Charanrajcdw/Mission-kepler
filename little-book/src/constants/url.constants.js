export const API_URLS = {
  blogs: "https://jsonmockserver.vercel.app/api/blogs",
  users: "https://jsonmockserver.vercel.app/api/users",
  images: "https://jsonmockserver.vercel.app/",
};
