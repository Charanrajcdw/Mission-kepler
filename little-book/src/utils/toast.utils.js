import { toast } from "react-toastify";

export const showSuccessToast = (message, currentTheme) => {
  toast.success(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};
