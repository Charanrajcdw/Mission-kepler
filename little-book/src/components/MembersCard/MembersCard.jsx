import PropTypes from "prop-types";
import styles from "./MembersCard.module.scss";
import Image from "../Image/Image";
import { API_URLS } from "../../constants";

const MembersCard = ({ name, photo, address }) => {
  return (
    <div className={styles.memberCard}>
      <Image src={`${API_URLS.images}${photo}`} alt={name} className="memberImg" />
      <p className={styles.memberName}>{name}</p>
      <span className={styles.memberCity}>{address.city}</span>
    </div>
  );
};

MembersCard.propTypes = {
  name: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
  address: PropTypes.shape({ city: PropTypes.string.isRequired }),
};

export default MembersCard;
