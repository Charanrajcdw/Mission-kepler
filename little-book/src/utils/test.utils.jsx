import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import ThemeContextProvider from "../contexts/theme.context";
import blogSlice from "../store/slices/blogSlice";
import { THEME } from "../constants";

const customRender = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
    ...options
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const customCreateSnapshot = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return renderer.create(<ProviderWrapper>{ui}</ProviderWrapper>).toJSON();
};

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render, customCreateSnapshot as createSnapshot };
