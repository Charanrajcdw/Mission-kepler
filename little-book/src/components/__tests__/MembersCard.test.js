import { fireEvent, render, screen, createSnapshot } from "../../utils/test.utils.jsx";
import MembersCard from "../MembersCard/MembersCard";
import BrokenImage from "../../assets/blank-image.jpeg";
import { API_URLS } from "../../constants";

describe("MembersCard", () => {
  it("should render members card", () => {
    const cardData = {
      name: "username",
      photo: "images/users/1.jpeg",
      address: { city: "user address" },
    };
    render(<MembersCard {...cardData} />);
    const nameElement = screen.getByText(cardData.name);
    const addressElement = screen.getByText(cardData.address.city);
    const imgElement = screen.getByRole("img");
    expect(nameElement).toBeInTheDocument();
    expect(addressElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src", `${API_URLS.images}${cardData.photo}`);
  });

  it("should match members card snapshot", () => {
    const cardData = {
      name: "username",
      photo: "images/users/1.jpeg",
      address: { city: "user address" },
    };
    const membersCard = createSnapshot(<MembersCard {...cardData} />);
    expect(membersCard).toMatchSnapshot();
  });

  it("should render card with broken image if no img src available", () => {
    const cardData = {
      name: "username",
      photo: "",
      address: { city: "user address" },
    };
    render(<MembersCard {...cardData} />);
    const nameElement = screen.getByText(cardData.name);
    const addressElement = screen.getByText(cardData.address.city);
    const imgElement = screen.getByRole("img");
    fireEvent.error(imgElement);
    expect(nameElement).toBeInTheDocument();
    expect(addressElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute("src", BrokenImage);
  });
});
