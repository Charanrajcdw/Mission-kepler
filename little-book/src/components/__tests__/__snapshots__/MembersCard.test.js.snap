// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersCard should match members card snapshot 1`] = `
<div
  className="memberCard"
>
  <div
    className="memberImg"
  >
    <img
      alt="username"
      className="img"
      onError={[Function]}
      src="https://jsonmockserver.vercel.app/images/users/1.jpeg"
    />
  </div>
  <p
    className="memberName"
  >
    username
  </p>
  <span
    className="memberCity"
  >
    user address
  </span>
</div>
`;
